chia/wallet/nft_wallet/nft_wallet.py

```diff
@@ -810,34 +811,33 @@ async def make_nft1_offer(
             offered_amount = uint64(abs(offer_dict[offered_asset_id]))
             royalty_amount = uint64(offered_amount * royalty_percentage / 10000)
             if offered_amount == royalty_amount:
+                # Due to the coin id's matching this is impossible unless we use separate coins
+                # when fulfilling the offer.
                 raise ValueError("Amount offered and amount paid in royalties are equal")
             if offered_asset_id is None:
                 # std xch offer
                 wallet = wallet_state_manager.main_wallet
+                coin_amount_needed: int = offered_amount + royalty_amount + fee
             else:
                 # cat offer
                 wallet = await wallet_state_manager.get_wallet_for_asset_id(offered_asset_id.hex())
-
-            if wallet.type() == WalletType.STANDARD_WALLET:
-                coin_amount_needed: int = offered_amount + royalty_amount + fee
-            else:
                 coin_amount_needed = offered_amount + royalty_amount
-            pmt_coins = list(await wallet.get_coins_to_offer(offered_asset_id, coin_amount_needed))
 
+            pmt_coins = list(await wallet.get_coins_to_offer(offered_asset_id, coin_amount_needed, min_coin_amount))
             notarized_payments = Offer.notarize_payments(requested_payments, pmt_coins)
             announcements_to_assert = Offer.calculate_announcements(notarized_payments, driver_dict)
             # Calculate the royalty announcement separately
-            announcements_to_assert.extend(
-                Offer.calculate_announcements(
-                    {
-                        offered_asset_id: [
-                            NotarizedPayment(royalty_address, royalty_amount, [royalty_address], requested_asset_id)
-                        ]
-                    },
-                    driver_dict,
-                )
+            royalty_announcement = Offer.calculate_announcements(
+                {
+                    offered_asset_id: [
+                        NotarizedPayment(royalty_address, royalty_amount, [royalty_address], requested_asset_id)
+                    ]
+                },
+                driver_dict,
             )
-
+            # add royalty announcement to the list of announcements to assert
+            announcements_to_assert.extend(royalty_announcement)
+            royalty_ph = royalty_announcement[0].origin_info
```

Comment:
I wrote this in a comment before some pushes but I don't like using the puzhash from one of these calculated announcements being used to set `royalty_ph`.  We should instead be constructing the puzzle hash ourselves `construct_puzzle`, the driver_dict, and `OFFER_MOD`.

Reply:
ah i misunderstood what you meant, gotcha!